src/services/PaymentService.ts

```diff
@@ -236,3 +254,4 @@ export class PaymentService {
 }
 
 class NotFoundSubscriptionError extends Error {}
+export class CouponAlreadyAppliedError extends Error {}
```

Comment:
Please add the error message here: 
```
export class CouponAlreadyAppliedError extends Error {
  constructor() {
     super('User already applied coupon');
     
     Object.setPrototypeOf(this, CouponAlreadyAppliedError.prototype);
  }
}
```

The line 
```
Object.setPrototypeOf(this, CouponAlreadyAppliedError.prototype);
```

Is due to a TypeScript error: https://dannyguo.medium.com/how-to-fix-instanceof-not-working-for-custom-errors-in-typescript-1df978100a27

If you do not put it, `instanceof` will not work